lib/input/http_server.go

```diff
@@ -510,7 +519,15 @@ func (h *HTTPServer) postHandler(w http.ResponseWriter, r *http.Request) {
 		}
 
 		if plen := responseMsg.Len(); plen == 1 {
-			payload := responseMsg.Get(0).Get()
+			part := responseMsg.Get(0)
+			part.Metadata().Iter(func(k, v string) error {
+				if h.metaFilter.Match(k) {
```

Comment:
Forgot to ask... Maybe I should do `h.metaFilter.Match(strings.ToLower(k))` to be consistent with the client? Now that I think about it, what is the reasoning for doing that in the client?

Reply:
I have a vague memory that this was because:

1. metadata keys created by benthos tend to be lowercase
2. headers are case insensitive so we can do what we want with them
3. I believe the Go client already normalizes headers in its own way (capitalizes them), so we're just overriding it with our own taste

I think your changes are fine as is as we're not bothered about normalizing the data that leaves Benthos (capital letters here will have been created by the user, so we can assume they're aware of them).

The patterns matched against metadata that we're adding ourselves from a response need to be tested against the keys the user is going to see in their data, so I'd say we leave that as well.